src/transport/raw/tests/TestTCP.cpp

```diff
@@ -382,15 +365,23 @@ int TestDataCallbackCheck(const uint8_t * message, size_t length, int count, voi
 
 } // namespace
 
-void chip::Transport::TCPTest::CheckProcessReceivedBuffer(nlTestSuite * inSuite, void * inContext)
+namespace chip {
+namespace Transport {
+class TCPTest
+{
+public:
+    static void CheckProcessReceivedBuffer(TestContext * ctx);
+};
+} // namespace Transport
+} // namespace chip
+void chip::Transport::TCPTest::CheckProcessReceivedBuffer(TestContext * ctx)
```

Comment:
Couldn't you make a normal test out of it using `TEST_F` and do not create this namespaced class?

Reply:
The original test author probably created the namespaced class to contain that function because it uses `TCPBase::ActiveConnectionState` which is a protected member of TCPBase (it wasn't my idea to do the namespacing, I just left it how it was originally written).  Taking the function out of that namespaced class and just making it a TEST_F yields the error `‘struct chip::Transport::TCPBase::ActiveConnectionState’ is protected within this context`.  I think the author put it into TCPTest to take advantage of the fact that TCPBase contains `friend class TCPTest;`

Reply:
Oh I see. In that case leave it as it is